ui/packages/atlasmap-core/src/utils/mapping-serializer.ts

```diff
@@ -292,6 +293,30 @@ export class MappingSerializer {
     return jsonMapping;
   }
 
+  static extractCheckVersion(json: IAtlasMappingContainer, cfg: ConfigModel) {
+    const currentVersion = cfg.initializationService.getUIVersion();
+    const mappingVersion = this.deserializeAtlasMappingVersion(json);
+    const mappingVersionComps = mappingVersion.split('.');
+    const currentVersionComps = currentVersion.split('.');
+    if (
+      (mappingVersion.length > 0 &&
+        (currentVersionComps.length < 2 || mappingVersionComps.length < 2)) ||
+      +currentVersionComps[0] < +mappingVersionComps[0] ||
+      +currentVersionComps[1] < +mappingVersionComps[1]
+    ) {
+      cfg.errorService.addError(
+        new ErrorInfo({
+          message: `Mappings file version mismatch.  Expected ${currentVersion}, detected ${mappingVersion}`,
+          level: ErrorLevel.ERROR,
+          scope: ErrorScope.APPLICATION,
+          type: ErrorType.USER,
+        })
+      );
+    } else {
+      cfg.mappings!.version = currentVersion;
```

Comment:
I think we can update the version even if the version mismatch is detected. Then the error is shown only at the first time, but if the deserialization succeeds, the version is overridden and the next adm export will create a compatible one. Let user decide if they continue with it.